OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -59,10 +59,14 @@ class TheWorld : public Component {
     void add(Component* comp) {
         // add it the property list of components that owns and serializes them
         updProperty_components().adoptAndAppendValue(comp);
+        addComponent(comp);
     }
 
     // Top level connection method for all encompassing Component
-    void connect() { Super::connect(*this); }
+    void connect() { 
+        initComponentTreeTraversal(*this);
+        Super::connect(*this);
```

Comment:
@aymanhab  `Super::` is unnecessary. No redesign/revamp necessary.
@chrisdembia This method could easily have been `setupWorld()` or any other method of TheWorld component. The `Super::do()` policy is for our or virtual `extend<something>()` methods. In this case the Super:: is unnecessary anyways.